tests/lib/rules/no-http-redirects/tests.ts

```diff
@@ -0,0 +1,107 @@
+import { IRuleTest } from '../../../helpers/rule-test-type';
+import { getRuleName } from '../../../../src/lib/utils/rule-helpers';
+import * as ruleRunner from '../../../helpers/rule-runner';
+import { generateHTMLPage } from '../../../helpers/misc';
+
+const ruleName = getRuleName(__dirname);
+
+const tests: Array<IRuleTest> = [
+    {
+        name: 'No redirects pass the rule',
+        serverConfig: generateHTMLPage()
+    },
+    {
+        name: 'Redirect in resource fails',
+        reports: [{ message: '2 redirects detected for http://localhost/image.png (max is 0).' }],
+        serverConfig: {
+            '/': generateHTMLPage('', '<img src="/image.png">'),
+            '/image.png': {
+                content: '/image2.png',
+                status: 302
```

Comment:
You should also use one of the other redirection status codes in the tests (e.g.: 301).